tests/utils.go

```diff
@@ -136,7 +136,7 @@ const (
 	WarningEvent EventType = "Warning"
 )
 
-const defaultTestGracePeriod int64 = 0
+const defaultTestGracePeriod int64 = 1
```

Comment:
by having this as 0 not it changes all the tests that uses it. Before it thought `isACPIEnabled` was true so function tests expect the status of `Failed` but with the change that status is not possible since `0` no longer returns that ACPI is enabled.

Reply:
which functional tests failed? It's unclear how this is possible.

I thought a grace period of 0 has functionally been treated the same both before and after your patch.  Looking at virt-handler, even if we pass the isACPIEnabled, the expired boolean is true which ultimately falls through to the kill vm function later on. 

https://github.com/kubevirt/kubevirt/blob/9147f1cc39608c07c523aaa3deab24f3c9729a89/pkg/virt-handler/vm.go#L1999 
```
        if isACPIEnabled(vmi, domain) {                                            
                expired, timeLeft := d.hasGracePeriodExpired(domain)
                if !expired {  
```

Reply:
The problem is later, that after being killed it can only have the status of success, instead of failed. https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L2594-L2596

The other way to fix this would be to assert the success status instead of failed, but I wasn't sure what the correct way was. If we wont to test that it updates to the correct failed status when it is no killed in time with graceperiod we need to set it to something other than 0. 

The test failing was `test_id:1652` in `vmi_lifecycle_test.go`

Reply:
Ah, i see

>The other way to fix this would be to assert the success status instead of failed, but I wasn't sure what the correct way was. If we wont to test that it updates to the correct failed status when it is no killed in time with graceperiod we need to set it to something other than 0.

So the best I can tell, this ACPI with grace period of `0` never worked as intended really. I still don't understand this previous logic was trying to do.

```
zero := int64(0)
return vmi.Spec.TerminationGracePeriodSeconds != &zero  && 
```

For instance, If i mess around with this logic, I can't make it do anything that makes sense.

```
        zero := int(0)
        newZero := int(0)
        nonZero := int(100)

        var pointer *int

        if pointer != &zero {
                fmt.Printf("nil pointer doesn't equal zero\n")
        }

        pointer = &nonZero
        if pointer != &zero {
                fmt.Printf("non zero pointer doesn't equal zero\n")
        }

        pointer = &newZero

        if pointer != &zero {
                fmt.Printf("zero pointer doesn't equal other zero pointer\n")
        }
```

I think this ACPI logic has been flawed and you fixed it. I think when ACPI is enabled but the grace period is zero, that the only expected outcome is that the domain will be destroyed, which to me means "success". 

If that one test is changed, will everything else pass?

Reply:
> If that one test is changed, will everything else pass?

What I meant by that is, if we return defaultTestGracePeriod = 0 and then change that one test to look for success instead of failure, will that restore the test suite to working condition... or are there more failures to contend with as a result of this ACPI change?